omnibus/files/private-chef-cookbooks/private-chef/libraries/preflight_solr_validator.rb

```diff
@@ -48,6 +66,172 @@ def verify_sane_reindex_sleep_times
+      an unsupported configuration.
+
+      To disable the internal Elasticsearch, add the following to #{CHEF_SERVER_CONFIG_FILE}:
+
+          elasticsearch['enable'] = false
+
+      To use the internal Elasticsearch, consider removing the configuration
+      entry for opscode_solr4['external'].
+
+      EOM
+    end
+
+    if solr_enabled? && external?
+      fail_with <<~EOM
+
+      The #{CHEF_SERVER_NAME} is configured to use an external search
+      index but the internal Solr is still enabled. This is an
+      unsupported configuration.
+
+      To disable the internal Solr, add the following to #{CHEF_SERVER_CONFIG_FILE}:
+
+          opscode_solr4['enable'] = false
+
+      Alternatively, if you are attempting to use the internal Solr,
+      add the following to #{CHEF_SERVER_CONFIG_FILE}:
+
+          opscode_solr4['enable'] = true
+          opscode_solr4['external'] = false
+
+      EOM
+    end
+  end
+
+  # The user might have a combination of solr, rabbitmq or
+  # opscode-expander explicitly enabled in their config.
+  #
+  # If we are using the new internal elasticsearch, this is an
+  # unsupported configuration.
+  #
+  # To test: setup standalone cs with 13.2.0 add
+  # opscode_solr4['enable'] = true
+  # rabbitmq['enable'] = true
+  # opscode_expander['enable'] = true to chef-server.rb
+  # then upgrade to 13.3
+  def verify_unused_services_are_disabled_if_using_internal_es
+    if elasticsearch_enabled?
+      if cs_solr_attr['enable']
+        fail_with <<~EOM
+
+      The #{CHEF_SERVER_NAME} is configured to use its internal
+      Elasticsearch installation, but opscode_solr4 has been manually
+      enabled. This is an unsupported configuration.
+
+      Please update #{CHEF_SERVER_CONFIG_FILE} to ensure that
+      opscode_solr4 is disabled:
+
+          opscode_solr4['enable'] = false
+
+      EOM
+      end
+
+      if cs_rabbitmq_attr['enable']
+        fail_with <<~EOM
+
+      The #{CHEF_SERVER_NAME} is configured to use its internal
+      Elasticsearch installation, but RabbitMQ has been manually
+      enabled. This is an unsupported configuration.
+
+      Please update #{CHEF_SERVER_CONFIG_FILE} to ensure that
+      rabbitmq is disabled:
+
+          rabbitmq['enable'] = false
+
+      EOM
+      end
+
+      if cs_opscode_expander_attr['enable']
+        fail_with <<~EOM
+
+      The #{CHEF_SERVER_NAME} is configured to use its internal
+      Elasticsearch installation, but opscode-expander has been
+      manually enabled. This is an unsupported configuration.
+
+      Please update #{CHEF_SERVER_CONFIG_FILE} to ensure that
+      opscode-expander is disabled:
+
+          opscode_expander['enable'] = false
+
+      EOM
+      end
+    end
+  end
+
+  def verify_no_explicit_external_disable_when_es_enabled
+    return if cs_solr_attr['external'].nil?
+
+    # TODO(ssd) 2020-05-13: Rather than this check, we could just
+    # ignore that configuration when parsing the configuration.
```

Comment:
Another alternative is to not overload the `external` config item like we are currently. But I'm not sure on the level of effort there.